### ISSUE TITLE
CC0021 generates a use NameOf for a constant defined as itself

### ISSUE THREAD

**Issue by CZEMacLeod** [code-cracker/code-cracker#925]
#Bug

CC0021 suggests using a constant definition name as itself which does not compile.

````vb
'the code that reproduces the bug
Class MyClass
    Public Const MyConst="MyConst"
End Class
````

Current output after fix applied:

````vb
'code fixed incorrectly
Class MyClass
    Public Const MyConst=NameOf(MyConst)
End Class
````

Error:
BC30500	Constant 'MyConst' cannot depend on its own value.

**Comment by carloscds** [comment]
@CZEMacLeod This analyzer checks if value of variable it´s equal your name and then suggest a fix;

**Comment by CZEMacLeod** [comment]
I'm not quite sure what you mean by that.
The problem is that CC0021 from the NuGet package version on NuGet.org in visual basic suggests replacing a text string value for a constant with the same name. If you accept this change, you get error BC30500 which means the code does not compile.

**Comment by carloscds** [comment]
@CZEMacLeod It's make sense, A solution it's ignore rule in variable declaration, when name and value are same. What do you think about ?

**Comment by CZEMacLeod** [comment]
Yeah - That is what is required. Do not offer CC0021 when the nameof operator would point to the item you are declaring.
